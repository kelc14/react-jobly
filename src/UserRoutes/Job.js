import React from "react";
import "./Job.css";
import { Link } from "react-router-dom";

const Job = () => {
  return (
    <div className="Job-container">
      <p className="Job-title"> Job Title Goes Here </p>
      <p className="Job-company"> Company Name Goes Here </p>
      <p className="Job-about"> Description: 1,0000000 </p>
      <p className="Job-about"> Number of Employees: 00000 </p>
      <Link to="/">APPLY NOW</Link>
    </div>
  );
};

export default Job;
