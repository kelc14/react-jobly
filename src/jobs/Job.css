.Job-container {
  width: 70vw;
  background-color: white;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0 auto;
  padding: 20px;
  overflow: hidden;
  text-align: left;
  position: relative;
  margin-bottom: 25px;
}

.Job-title {
  text-align: left;
  font-weight: bold;
  font-size: 1.2rem;
}
.Job-company {
  text-align: left;
  font-weight: bold;
  font-size: 1rem;
}

.Job-about {
  text-align: left;
  font-size: 0.8rem;
}

.Job-container a {
  position: absolute;
  right: 0;
  bottom: 0;
  background-color: goldenrod;

  padding: 10px;
  margin: 20px;
  border-radius: 8px;
  text-decoration: none;
  color: white;
  font-weight: 500;
  /* text-shadow: 2px 7px 5px rgba(0, 0, 0, 0.3),
    0px -4px 10px rgba(255, 255, 255, 0.3); */
}

.Job-container a:hover {
  background-color: white;
  border: 1px solid goldenrod;
  color: goldenrod;
}
