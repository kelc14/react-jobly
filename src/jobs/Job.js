import React, { useEffect, useState } from "react";
import "./Job.css";
import { Link } from "react-router-dom";

const Job = ({ jobData, companiesData }) => {
  const [company, setCompany] = useState({});
  const { title, salary, equity, company_handle } = jobData;

  useEffect(() => {
    if (typeof companiesData === "Array") {
      let companyData = companiesData.find(
        (comp) => comp.handle === company_handle
      );
      setCompany({ ...companyData });
    } else {
      setCompany(companiesData);
    }
  }, []);

  return (
    <div className="Job-container">
      <p className="Job-title"> {title} </p>
      <p className="Job-company"> {company.name} </p>
      <p className="Job-about"> Salary: {salary} </p>
      <p className="Job-about"> Equity: {equity} </p>
      <Link to="/">APPLY NOW</Link>
    </div>
  );
};

export default Job;
