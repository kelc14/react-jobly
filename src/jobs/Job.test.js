import UserContext from "../hooks/UserContext";
import { render } from "@testing-library/react";
import Job from "./Job";

const customRender = (ui, { providerProps, ...renderOptions }) => {
  return render(
    <UserContext.Provider value={providerProps}>{ui}</UserContext.Provider>,
    renderOptions
  );
};

let providerProps;
let job;
let company;
beforeEach(() => {
  providerProps = {
    username: "testingUser",
    firstName: "testFirst",
    lastName: "testLast",
    email: "email@email.com",
    isAdmin: false,
    jobs: [null],
  };
  job = {
    id: 1,
    title: "Test job",
    salary: 110000,
    equity: "0",
    company_handle: "test-company",
  };
  company = {
    name: "Test Company",
    company_handle: "test-company",
  };
});

/**
 * SMOKE TEST
 */
test("Job renders without crashing", () => {
  customRender(<Job jobData={job} companiesData={company} />, {
    providerProps,
  });
});

/**
 * SNAPSHOT TEST
 */
test("Job matches snapshot", () => {
  const { asFragment } = customRender(
    <Job jobData={job} companiesData={company} />,
    {
      providerProps,
    }
  );
  expect(asFragment()).toMatchSnapshot();
});
