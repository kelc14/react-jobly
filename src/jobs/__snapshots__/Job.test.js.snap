// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Job-container"
  >
    <p
      class="Job-title"
    >
       Test job 
    </p>
    <p
      class="Job-company"
    >
       Test Company 
    </p>
    <p
      class="Job-about"
    >
       Salary: 110000 
    </p>
    <p
      class="Job-about"
    >
       Equity: 0 
    </p>
    <form>
      <button
        class="Job-apply-btn"
      >
        APPLY NOW
      </button>
    </form>
  </div>
</DocumentFragment>
`;
